pkg/vault/contracts/BatchRouter.sol

```diff
@@ -526,8 +528,10 @@ contract BatchRouter is IBatchRouter, BatchRouterCommon, ReentrancyGuardTransien
                         // Refund unused portion of BPT flashloan to the Vault.
                         if (bptAmountIn < stepMaxAmountIn) {
                             uint256 refundAmount = stepMaxAmountIn - bptAmountIn;
-                            stepTokenIn.safeTransfer(address(_vault), refundAmount);
-                            _vault.settle(stepTokenIn, refundAmount);
+                            if (refundAmount > 0) {
```

Comment:
This is always the case, as we're already checking that `stepMaxAmountIn > bptAmountIn` above. We can remove this `if` safely.

Reply:
Ooof; I just blindly did some of these without checking whether I needed to...